rosbag2_examples/rosbag2_examples_py/rosbag2_examples_py/simple_bag_recorder.py

```diff
@@ -53,9 +54,13 @@ def topic_callback(self, msg):
 
 def main(args=None):
     rclpy.init(args=args)
-    sbr = SimpleBagRecorder()
-    rclpy.spin(sbr)
-    rclpy.shutdown()
+    try:
+        sbr = SimpleBagRecorder()
+        rclpy.spin(sbr)
+    except KeyboardInterrupt:
+        pass
+    except ExternalShutdownException:
+        rclpy.shutdown()
```

Comment:
this is required to avoid the following trace stack with `Ctrl-C` to stop the executable.

```bash
Traceback (most recent call last):
  File "/root/ros2_ws/colcon_ws/install/rosbag2_examples_py/lib/rosbag2_examples_py/simple_bag_reader", line 33, in <module>
    sys.exit(load_entry_point('rosbag2-examples-py', 'console_scripts', 'simple_bag_reader')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/ros2_ws/colcon_ws/build/rosbag2_examples_py/rosbag2_examples_py/simple_bag_reader.py", line 49, in main
    rclpy.spin(sbr)
  File "/root/ros2_ws/colcon_ws/install/rclpy/lib/python3.12/site-packages/rclpy/__init__.py", line 240, in spin
    executor.spin_once()
  File "/root/ros2_ws/colcon_ws/install/rclpy/lib/python3.12/site-packages/rclpy/executors.py", line 852, in spin_once
    self._spin_once_impl(timeout_sec)
  File "/root/ros2_ws/colcon_ws/install/rclpy/lib/python3.12/site-packages/rclpy/executors.py", line 836, in _spin_once_impl
    handler, entity, node = self.wait_for_ready_callbacks(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/ros2_ws/colcon_ws/install/rclpy/lib/python3.12/site-packages/rclpy/executors.py", line 806, in wait_for_ready_callbacks
    return next(self._cb_iter)
           ^^^^^^^^^^^^^^^^^^^
  File "/root/ros2_ws/colcon_ws/install/rclpy/lib/python3.12/site-packages/rclpy/executors.py", line 710, in _wait_for_ready_callbacks
    wait_set.wait(timeout_nsec)
```

Reply:
@fujitatomoya I'm wondering if it makes sense to add the section `finally` with `rclpy.shutdown()` as well?

Reply:
thanks for the comment. if we use `finally` with `rclpy.shutdown()`, the following exception will be generated, because rclpy has been gracefully shut down by sigint. besides https://github.com/ros2/examples/tree/rolling/rclpy does not have `finally` either, so maybe we can keep this as it is.

```bash
rclpy._rclpy_pybind11.RCLError: failed to shutdown: rcl_shutdown already called on the given context, at /root/ros2_ws/colcon_ws/src/ros2/rcl/rcl/src/rcl/init.c:333
```